src/LogFileReader.cpp

```diff
@@ -75,11 +78,37 @@ bool LogFileReader::readAll(const QString& fileName)
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
     // TODO implement this first
+    QStringList lines = line.split(',');
+    bool ok;
+    if(lines.count()!=3)
+        return false;
+    QStringList lines_1 = lines[0].split(':');
+    QStringList lines_2 = lines[1].split('.');
+    int value = lines_1[0].toInt(&ok);
+    if((!ok)||(value>24)||(value<0))
+        return false;
```

Comment:
Instead of parsing the time yourself, I'd recommend using QTime's `fromString` function. Here is a link to the documentation

http://doc.qt.io/qt-5/qtime.html#fromString-1

You can use the `isValid()` function afterwards to check if the conversion was successful.